packagedb/tests/test_api.py

```diff
@@ -1179,3 +1179,11 @@ def test_api_purl_validation_empty_request(self):
         }
 
         self.assertAlmostEquals(expected, response1.data)
+
+
+class ToGolangPurlTestCase(TestCase):
+
+    def test_to_golang_purl(self):
+        response = self.client.get("/api/to_purl/go", data={"go_package": "github.com/gorilla/mux@v1.7.3"}, follow=True)
```

Comment:
AFAIK, `github.com/gorilla/mux@v1.7.3` is not a valid import statement.
Does Go allow specifying a particular package version using an `@` symbol in an import statement?

Reply:
Good catch, not a go import https://github.com/moby/moby/blob/6c10086976d07d4746e03dcfd188972a2f07e1c9/vendor.mod#L51 but `*.mod` files does have version representation like this? should I change the code in packageurl-python to accept this kind of inputs as well ?

Reply:
> should I change the code in packageurl-python to accept this kind of inputs as well ?

Update the code to also accept this would make sense. With proper documentation.

Reply:
BTW, see also https://stackoverflow.com/questions/24855081/how-do-i-import-a-specific-version-of-a-package-using-go-get and in particular the reference to http://labix.org/gopkg.in

Reply:
> Good catch, not a go import https://github.com/moby/moby/blob/6c10086976d07d4746e03dcfd188972a2f07e1c9/vendor.mod#L51 but `*.mod` files does have version representation like this? should I change the code in packageurl-python to accept this kind of inputs as well ?

IMO it makes sense to support getting PURLs from both Go import and Go Manifest.
However, I'm not sure if we want to support obtaining PURLs from Go dependency management commands like...
```bash 
go get github.com/gorilla/mux@v1.7.3
```

Reply:
I have added a PR here https://github.com/package-url/packageurl-python/pull/148/files, please have a look

Reply:
> @keshav-space so should we throw an error when we encounter any package like this ? when there is "@" in package, we should error out ?

If we don't support them in packageurl-python, then we should raise an error here.

Reply:
I think this is all OK and handled in the packageurl library